pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -510,15 +587,22 @@ func (nc NodeController) tryUpdateNodeStatus(node *api.Node) (error, time.Durati
 				nc.recordNodeOfflineEvent(node)
 			}
 		}
-		_, err = nc.kubeClient.Nodes().Update(node)
-		if err != nil {
-			glog.Errorf("Error updating node %s: %v", node.Name, err)
-		} else {
-			return nil, gracePeriod, lastReadyCondition
+		if !api.Semantic.DeepEqual(readyCondition, lastReadyCondition) {
```

Comment:
readyCondition can be modified in the if's above. But there's other problem here - I should look at condition from node.Status, not saved readyCondition. And I can't put it in readyCondition as we use the fact that it's nil in Monitor...() method...